WordPress/src/main/java/org/wordpress/android/util/PerAppLocaleManager.kt

```diff
@@ -31,86 +31,26 @@ class PerAppLocaleManager @Inject constructor(
-     */
-    fun checkAndUpdateOldLanguagePrefKey() {
-        val prefKey = LocaleManager.getLocalePrefKeyString()
-        val inAppLanguage = appPrefsWrapper.getPrefString(prefKey, "")
-        val perAppLanguage = getCurrentLocale().language
-        if (perAppLanguage.isNotEmpty() && inAppLanguage.equals(perAppLanguage).not()) {
-            appPrefsWrapper.setPrefString(prefKey, perAppLanguage)
-            appLogWrapper.d(
-                AppLog.T.SETTINGS,
-                "PerAppLocaleManager: changed inAppLanguage from $inAppLanguage to $perAppLanguage"
-            )
-        }
-    }
-
-    /**
-     * This routine can be helpful during development to reset the app locale
-     */
-    @Suppress("unused")
-    fun resetApplicationLocale() {
-        AppCompatDelegate.setApplicationLocales(LocaleListCompat.getEmptyLocaleList())
-    }
-
     private fun setCurrentLocaleByLanguageCode(languageCode: String) {
         // We shouldn't have to replace "_" with "-" but this is in order to work with our existing language picker
         // on pre-Android 13 devices
         val appLocale = LocaleListCompat.forLanguageTags(languageCode.replace("_", "-"))
         AppCompatDelegate.setApplicationLocales(appLocale)
-        checkAndUpdateOldLanguagePrefKey()
     }
 
     /**
-     * Previously the app locale was stored in SharedPreferences, so here we migrate to AndroidX per-app language prefs
+     * Previously the app locale was stored in SharedPreferences, so here we migrate to AndroidX per-app language prefs.
+     * This was added in our Jan 2025 release and can be removed after a few subsequent releases.
      */
     fun performMigrationIfNecessary() {
-        if (isApplicationLocaleEmpty()) {
-            val prefKey = LocaleManager.getLocalePrefKeyString()
-            val previousLanguage = appPrefsWrapper.getPrefString(prefKey, "")
-            if (previousLanguage?.isNotEmpty() == true) {
-                appLogWrapper.d(
-                    AppLog.T.SETTINGS,
-                    "PerAppLocaleManager: performing migration to AndroidX per-app language prefs"
-                )
-                setCurrentLocaleByLanguageCode(previousLanguage)
-            } else {
-                appLogWrapper.d(
-                    AppLog.T.SETTINGS,
-                    "PerAppLocaleManager: setting default locale"
-                )
-                setCurrentLocaleByLanguageCode(Locale.getDefault().language)
-            }
-        } else {
-            checkAndUpdateOldLanguagePrefKey()
+        val previousLanguage = appPrefsWrapper.getPrefString(OLD_LOCALE_PREF_KEY_STRING, "")
+        if (previousLanguage?.isNotEmpty() == true) {
+            appLogWrapper.d(
+                AppLog.T.SETTINGS,
+                "PerAppLocaleManager: performing migration to AndroidX per-app language prefs"
+            )
+            setCurrentLocaleByLanguageCode(previousLanguage)
+            appPrefsWrapper.removePref(OLD_LOCALE_PREF_KEY_STRING)
```

Comment:
Question (❓): This prefs removal wouldn't happen for users that are already migrated via #21504 and [this call](https://github.com/wordpress-mobile/WordPress-Android/pull/21504/files#diff-49dc0f74afe58f57b2d6b8640d70df34081db077e00b559ee817fc79e5e0ed60R370) being introduced back then and already live, right? 🤔

Reply:
That's correct but I believe it's mostly harmless.

Reply:
So, to make sure I understand it fully, those users that are already migrated with the current production version of the app will not have this pref removed, only existing users that haven't yet updated to that version would eventually have this pref removed, when they get this new app update, right? 😊

Reply:
For those users who have already been migrated, this pref would still be removed because the check for `isNotEmpty`. In other words, we do the migration if the pref exists, which will remove the pref. This does mean that users who were previously migrated would be migrated again, but I don't see any harm in that.

Reply:
Ah cool, totally got it now Nick, thanks! 🙇